### PR TITLE
Refactor NpTicket parser and add NpTicket signature documentation

### PR DESCRIPTION
**Pull request by Slendy**
Makes ticket body parsing and ticket signature verification more abstract and modular rather than having everything done in a single file.

### CODE REVIEW

File: ProjectLighthouse.Tests/Unit/TicketTests.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+public class TicketTests
+{
+    [Fact]
+    public void CanReadTicket()
+    {
+        TicketBuilder builder = new TicketBuilder().SetCountry("br")
+            .SetUserId(21)
+            .SetDomain("us")
+            .SetStatus(0)
+            .SetIssuerId(0x74657374)
```
Comment (sudokoko) [inline ProjectLighthouse.Tests/Unit/TicketTests.cs:18]:
```suggestion
    public void CanReadTicket()
    {
        TicketBuilder builder = new TicketBuilder()
            .SetCountry("br")
            .SetUserId(21)
            .SetDomain("us")
            .SetStatus(0)
            .SetIssuerId(0x74657374)
```
Can this be moved down to be consistent with styling?

**Review by sudokoko — APPROVED** [review]
Looks good functionally, Styling nitpick

File: ProjectLighthouse.Tests/Unit/TicketTests.cs (hunk 1 above)
Comment (Slendy) [reply ProjectLighthouse.Tests/Unit/TicketTests.cs:18]: I can, but any time the file is formatted Rider tries to move it back

Comment (sudokoko) [reply ProjectLighthouse.Tests/Unit/TicketTests.cs:18]: Ah, that's alright then.

**Pull request merged**